src/crimson/os/seastore/cache.h

```diff
@@ -464,6 +464,20 @@ class Cache {
     return view->is_data_stable();
   }
 
+  bool is_extent_retired_by_trans(CachedExtent& extent, Transaction &t) const {
```

Comment:
The extent might be pending

Reply:
How about making this case explicit (instead of relying on the implicit correctness), e.g.
```c++
if (!extent->is_stable()) {
  if (extent->is_valid()) {
    // probably assert extent is local to transaction t ?
    return false;
  } else {
    return true;
  }
} else {
  ... (the stable case) ...
}
```